mula/scheduler/storage/filters/functions.py

```diff
@@ -27,11 +27,25 @@ def apply_filter(entity: DeclarativeBase, query: Query, filter_request: FilterRe
     for operator in filter_request.filters:
         expressions = []
         for filter_ in filter_request.filters[operator]:
+            if not hasattr(entity, filter_.column):
+                raise FilterError(f"Invalid filter field: {filter_.column} (error: not found)")
+
+            # If the filter field is not specified we will use the column name
+            # as the filter_field
             filter_field = filter_.field if filter_.field else filter_.column
 
             # Return the selected attribute of the model, e.g. Model.selected_attr
             entity_attr = getattr(entity, filter_.column)
 
+            # Check if the column we are filtering on is a relationship, if so
+            # we need to join the related entity to the query.
+            if hasattr(entity_attr, "property") and isinstance(entity_attr.property, RelationshipProperty):
+                related_entity = entity_attr.property.mapper.class_
+                query = query.join(related_entity)
```

Comment:
From here this check is repeated a few times. I suggest implementing a helper function:
```python3
def is_relationship_property(attr) -> bool:
    return hasattr(attr, "property") and isinstance(attr.property, RelationshipProperty)
```
(not sure what `attr` type should be exactly)

```suggestion
            if is_relationship_property(entity_attr):
                related_entity = entity_attr.property.mapper.class_
                query = query.join(related_entity)
```